packages/wcg-url/index.d.ts

```diff
@@ -0,0 +1,8 @@
+/// <reference lib="dom" />
+
+// typings being weird
+export class WCGURL implements URL {
```

Comment:
Do you need to fill in all methods from the URL interface? VS Code can fill in all missing methods and you can write `declare` in front of them to avoid writing the implementations.